models/activity.js

```diff
@@ -0,0 +1,56 @@
+// Import mongoose library
+const mongoose = require("mongoose");
+
+/* 
+Mongoose schema definition. See - https://mongoosejs.com/docs/guide.html#definition
+
+Defines the Activity.reference schema which consist of text and url. 
+
+Text: The text value is of type string, is required when the url value has characters of one or more. 
+
+URL: The url value is of type string and is not required, but when included should be a valid url 
+
+*/
+const referenceSchema = new mongoose.Schema({
+  text: {
+    type: String,
+    required: function () {
+      return this.url.length > 0;
+    },
+  },
+  url: {
+    type: String,
+    required: false,
+    validate(value) {
+      return /^(ftp|http|https):\/\/[^ "]+$/.test(value);
+    },
+  },
+});
+
+/* 
+Defines activity schema to consist of a description value that is of type string, it will be required, 
+
+trim any whitespace and will have a default of less than 250 characters. 
+*/
+const activitySchema = new mongoose.Schema({
+  description: {
+    type: String,
+    required: true,
+    trim: true,
+    validate(value) {
+      if (value.length > 250) {
+        throw new Error(
+          "Description is too long. Must be less than 250 characters."
+        );
+      }
+    },
+  },
```

Comment:
USE the validation `maxLength` key instead.

Reply:
updated